libcaf_core/caf/detail/typed_actor_util.hpp

```diff
@@ -146,4 +155,25 @@ struct is_normalized_signature<result<Out...>(In...)> {
 template <class F>
 constexpr bool is_normalized_signature_v = is_normalized_signature<F>::value;
 
+template <class... Sigs>
+struct are_signatures_normalized;
+
+template <class... Sigs>
+struct are_signatures_normalized<type_list<Sigs...>> {
+  static constexpr bool value
+    = (detail::is_normalized_signature_v<Sigs> && ...);
+};
+
+template <class... Sigs>
+constexpr bool are_signatures_normalized_v
+  = are_signatures_normalized<Sigs...>::value;
```

Comment:
If we're passing a `type_list`, then then the class only has a single template parameter.

```suggestion
template <class SigsList>
struct are_signatures_normalized;

template <class... Sigs>
struct are_signatures_normalized<type_list<Sigs...>> {
  static constexpr bool value
    = (detail::is_normalized_signature_v<Sigs> && ...);
};

template <class SigsList>
inline constexpr bool are_signatures_normalized_v
  = are_signatures_normalized<SigsList>::value;
```